scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/keras/models/Topology.scala

```diff
@@ -472,6 +473,70 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
     this.fit(x, batchSize, nbEpoch, null)
   }
 
+  def unwrapVectorAsNecessary(colType: DataType): (Row, Int) => Any = {
+    // to support both ML Vector and MLlib Vector
+    if (colType.typeName.contains("vector")) {
+      (row: Row, index: Int) => getVectorSeq(row, colType, index)
+    } else {
+      (row: Row, index: Int) => row.get(index)
+    }
+  }
+
+  private def getDataSet(
+      dataFrame: DataFrame,
+      batchSize: Int,
+      featureCol: String,
+      labelCol: String): FeatureSet[MiniBatch[T]] = {
```

Comment:
featureCol and labelCol should be a list of strings, similar to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/src/bigdl/orca/learn/pytorch/pytorch_spark_estimator.py#L68

Reply:
In scala these two are `Array[String]` currently